### PR TITLE
refactor engine/overworld/decorations.asm and engine/overworld/events.asm to pokecrystal

### PR DESCRIPTION
**Pull request by itsdarsh**
It's been a while since I've looked at this but I believe this is a refactor predominantly of engine/overworld/decorations.asm and engine/overworld/events.asm, in addition to related constants, macros and data. I also tried getting started on landmarks.asm but updating those constants is a little more involved than the simple sed macro I've been using, so that's where I stopped, and is why there may be constants that do not yet have a purpose.  My goal at the end of this process is to go through all the constants, figure out which ones are not used in PC even if they've been ported from pokecrystal, and remove them accordingly.

### CODE REVIEW

**Comment by Rangi42** [comment]
(Note to self, squash and merge this after review so the PR name is the commit message)

File: constants/landmark_constants.asm
Comment (Rangi42) [inline constants/landmark_constants.asm:10]: Don't prefix this `LANDMARK_` unless all the rest are too.

File: data/maps/blocks.asm
Comment (Rangi42) [inline data/maps/blocks.asm:447]: I'll wait for this TODO to merge, then.

File: data/maps/landmarks.asm
Comment (Rangi42) [inline data/maps/landmarks.asm:7]: If the `landmark` macro hasn't been modified to shift its coordinates by 8/16, then `dbbw` isn't necessary here.

File: data/maps/maps.asm
Comment (Rangi42) [inline data/maps/maps.asm:4]: Can you please go ahead and rename `ENVIRONMENT_5` to `ENCLOSED`? It was called `SHIP` in the original source, and behaves for day-night "outdoor" environments that you can't Fly/Dig/Teleport/Escape from. A shipdeck would work; I use it for rooftops as well.

File: engine/events/pokecenter_pc.asm
Comment (Rangi42) [inline engine/events/pokecenter_pc.asm:236]: Space alignment

File: engine/overworld/events.asm
Comment (Rangi42) [inline engine/overworld/events.asm:117]: Yes please.

File: data/maps/landmarks.asm
Comment (Rangi42) [reply data/maps/landmarks.asm:7]: (Updating the macro and all the coords would be nice, to match the Tilemap Studio status bar, but rust reverting this would be OK.)

File: data/maps/maps.asm
Comment (Rangi42) [reply data/maps/maps.asm:4]: (Noted for a future PR.)

**Review by Rangi42 — CHANGES_REQUESTED** [review]
Great work, thanks! Just a few changes as commented.

File: data/maps/blocks.asm
Comment (itsdarsh) [reply data/maps/blocks.asm:447]: How did I forget about this it's literally a sed lol

Comment (itsdarsh) [reply data/maps/blocks.asm:447]: Oh right because it changes the md5 hash

**Comment by itsdarsh** [comment]
I believe that should be everything

File: data/maps/blocks.asm
Comment (Rangi42) [inline data/maps/blocks.asm:447]: No need for the `TODO` comment if it's done ;)

File: constants/landmark_constants.asm
Comment (Rangi42) [reply constants/landmark_constants.asm]: This is still `LANDMARK_SPECIAL`.

File: data/maps/landmarks.asm
Comment (Rangi42) [reply data/maps/landmarks.asm]: This is still `dbbw`.

File: engine/events/pokecenter_pc.asm
Comment (Rangi42) [reply engine/events/pokecenter_pc.asm]: This is still misaligned.

**Pull request merged**
